autoload/clang_format.vim

```diff
@@ -203,8 +214,13 @@ function! s:detect_style_file() abort
     return findfile('.clang-format', dirname.';') != '' || findfile('_clang-format', dirname.';') != ''
 endfunction
 
-function! clang_format#format(line1, line2) abort
-    let args = printf(' -lines=%d:%d', a:line1, a:line2)
+" clang_format#format is were the magic happends.
+" ranges is a list of pairs, like [[start1,end1],[start2,end2]...]
+function! clang_format#format(ranges) abort
```

Comment:
This is public API. Can you avoid changing signature? Instead, please consider to make a new function like `clang_format#format_ranges`. `clang_format#format` would be able to call `clang_format#format_ranges` in its implementation